File: 1971-find-if-path-exists-in-graph/1971-find-if-path-exists-in-graph.cs
public class Solution {
    bool [] visited ;
    List <IList<int>> graph ;
    public bool ValidPath(int n, int[][] edges, int source, int destination) {
        // return false /true 
        // bi directinal 
        visited= new bool[n];
        graph = new List <IList<int>> ();
        // connected map;
        for( int i =0;i<n;i++){
             graph.Add(new List<int>());
          
            
        }
        foreach ( var edge in edges){
            int u = edge[0];
            int v = edge[1];
            graph[u].Add(v);
            graph[v].Add(u);// connected map complete 
        }
        return DFS ( source , destination);
        
    }
    bool DFS ( int node  , int destination){
        visited[node]= true;
        if ( node == destination ){
            return true ;
        }
        foreach ( int nei in graph[node]){
            if ( !visited[nei]){
                if (DFS ( nei, destination )){
                    return true;
                }
            }
        }
        return false;
    }
}

File: 1971-find-if-path-exists-in-graph/1971-find-if-path-exists-in-graph.java
class Solution {
     boolean [] visited ;
    List <List <Integer >> graph ;
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        visited = new boolean [n];
        graph = new ArrayList<>();
        for ( int i =0;i<n;i++){
            graph.add (new ArrayList<>());
        }
        for( int [] edge : edges){
            int u = edge[0];
            int v = edge [1];
            graph.get(u).add(v);
            graph.get(v).add(u);
        }
        return DFS ( source , destination );
        
    }
    boolean DFS ( int node , int destination ){
        visited[node]= true;
        if ( node== destination ){
            return true ;
        }
        for ( int nei:graph.get(node)){
            if (!visited[nei]){
                if(DFS(nei, destination )){
                    return true ;
                }

            }
        }
        return false;
    }
}